### ISSUE TITLE
When adding Microsoft.AspNetCore.Mvc.Testing to test project, dotnet publish --no-build command fails with NETSDK1085

### ISSUE THREAD

**Issue by Maxwellwr** [dotnet/aspnetcore#40330]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I have 2 projects. 
1. WebAPI application 
2. Unit test project. 

The test project uses WebApplicationFactory, for which the Microsoft.AspNetCore.Mvc.Testing package version 6.0.2 was included. Also in the test project, the IsPublishable property is set to false.

Then a sequence of commands is executed:
dotnet restore
dotnet build -c Release --no-restore
dotnet test -c Release --no-build
dotnet publish -c Release --no-build -o app/

As a result, I get the error "NETSDK1085: The 'NoBuild' property was set to true but the 'Build' target was invoked."

### Expected Behavior

WebAPI project is published without errors

### Steps To Reproduce

Clone repo:
https://github.com/Maxwellwr/net6publish

Execute:
dotnet restore
dotnet build -c Release --no-restore
dotnet test -c Release --no-build
dotnet publish -c Release --no-build -o app/

### .NET Version

6.0.200

**Comment by ltjax** [comment]
We have this problem as well, and I can narrow it down a little bit: It still works with Microsoft.AspNetCore.Mvc.Testing v5.0.17, but I get NETSDK1085 when I upgrade so 6.0.0

**Comment by areef-wk** [comment]
> We have this problem as well, and I can narrow it down a little bit: It still works with Microsoft.AspNetCore.Mvc.Testing v5.0.17, but I get NETSDK1085 when I upgrade so 6.0.0

I've encountered this too. The issue is still reproducible.

**Comment by yodasad** [comment]
I've encountered this too. The issue is still reproducible.
I do not want to publish test projects, so IsPublishable property is set to false.

**Comment by yodasad** [comment]
Overriding the `_GeneratePublishTestManifest` target (for instance, in a Directory.Build.targets) by adding the `'$(IsPublishable)'=='true' `condition fixes the issue:

 ```
 <Target Name="_GeneratePublishTestManifest" Condition="'$(IsPublishable)'=='true'" BeforeTargets="Publish" DependsOnTargets="_ResolveMvcTestProjectReferences">
    <ItemGroup>
      <_PublishManifestProjects Include="%(_ContentRootProjectReferences.FusionName)">
        <ContentRoot>~</ContentRoot>
      </_PublishManifestProjects>
      <_DepsFileToPublish Include="$([System.IO.Path]::ChangeExtension('%(_ContentRootProjectReferences.Identity)', 'deps.json'))" />
    </ItemGroup>

    <GenerateMvcTestManifestTask ManifestPath="$(PublishDir)MvcTestingAppManifest.json" Projects="@(_PublishManifestProjects)" />
    <Copy SourceFiles="%(_DepsFileToPublish.FullPath)" DestinationFolder="$(PublishDir)" Condition="Exists('%(_DepsFileToPublish.FullPath)')" />
  </Target>
```

It seems like the issue is fixed in 7.0: https://github.com/dotnet/aspnetcore/pull/43329.
The commit details are here: https://github.com/dotnet/aspnetcore/pull/43329/commits/e7f8f691b7c3d1337d6e8e66cd1ead24917b9aab

**Comment by captainsafia** [comment]
Closing as this has been backported to both 7.0 and 6.0. If the issue still persists, please file a new issue and I'll let you know.
